### PR TITLE
boundary constraint logic fix

### PR DESCRIPTION
**Pull request by pearcetm**
Fixes the behavior described and shown in a video here: https://github.com/openseadragon/openseadragon/pull/2239#issuecomment-1329849155

### CODE REVIEW

File: src/viewport.js [hunk 1]
```diff
@@ -514,7 +514,7 @@ $.Viewport.prototype = {
      * @param {OpenSeadragon.Rect} bounds
      * @returns {OpenSeadragon.Rect} constrained bounds.
      */
-    _applyBoundaryConstraints: function(bounds) {
+     _applyBoundaryConstraints: function(bounds) {
```
Comment (iangilman) [inline src/viewport.js:517]: This is a super tiny nitpick, but it seems this line got a space added at the beginning.

**Review by iangilman — COMMENTED** [review]
@pearcetm Thank you for the additional description and the video. This is definitely an improvement! I feel like we wrestled with this code for a good while in #2160; it's wild you could make it better by just deleting some code. 😆 

I've played around with this a bit and it seems solid. @joedf Do you have any thoughts on this patch? Anything we should be testing before merging it?

**Comment by joedf** [comment]
Works flawlessly, I'd say even much better! I recommend merging it and including this as is.
Only a minor thing is adding an extra empty line above line 536 for consistent formatting.

@pearcetm @iangilman Thanks for getting this fixed and verified! This whole constraints thing has definitely been a bit of a headache for me. I'm so glad, I knew it could be shorter, but just too confused. hahah 😬👍

**Review by iangilman — APPROVED** [review]
Awesome! Thank you for trying it out @joedf, and of course, thank you for making it happen, @pearcetm! Let's merge this! :)

**Pull request merged**